zarr/core.py

```diff
@@ -1800,7 +1813,7 @@ def _set_selection(self, indexer, value, fields=None):
             check_array_shape('value', value, sel_shape)
 
         # iterate over chunks in range
-        if not hasattr(self.store, "setitems") or self._synchronizer is not None \
+        if not hasattr(self.chunk_store, "setitems") or self._synchronizer is not None \
```

Comment:
I have to wonder if this wouldn't ever need to be the transformed_chunk_store. In general, this chain of replacement stores feels slightly like an anti-pattern.

Reply:
Yes, I'm also not particularly happy with this pattern, but also couldn't come up with a better solution for now. Happy about any ideas :+1: